esphome/components/wifi/__init__.py

```diff
@@ -179,8 +179,9 @@ def wifi_network_ap(value):
 def validate_variant(_):
     if CORE.is_esp32:
         variant = get_esp32_variant()
-        if variant in NO_WIFI_VARIANTS:
-            raise cv.Invalid(f"{variant} does not support WiFi")
+        integrations = CORE.loaded_integrations
+        if variant in NO_WIFI_VARIANTS and "esp32_hosted" not in integrations:
```

Comment:
Since this is in FINAL_VALIDATE_SCHEMA, it should use `fv.full_config.get()`
```suggestion
        if variant in NO_WIFI_VARIANTS and "esp32_hosted" not in fv.full_config.get():
```